### PR TITLE
(DOCSP-21722): Debug and Test React Native

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Add page on debugging and testing React Native. mostly refactor of content from the SDK readme

### Jira

- https://jira.mongodb.org/browse/DOCSP-21722

### Staged Changes (Requires MongoDB Corp SSO)

- [Debug and Test (React Native)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-21722/sdk/react-native/debug-and-test/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/test-and-debug.txt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+and `Mocha <https://mochajs.org/>`__.
+
+.. seealso:: 
+
+   `Official React Native Testing Documentation <https://reactnative.dev/docs/testing-overview>`__
+
+Clean Up Tests
+~~~~~~~~~~~~~~
+
+When testing the Realm React Native SDK, you must with
```
Comment (cbush) [inline source/sdk/react-native/test-and-debug.txt:30]:
```suggestion
When testing the Realm React Native SDK, you must close realms with 
```

File: source/sdk/react-native/test-and-debug.txt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+   `Official React Native Testing Documentation <https://reactnative.dev/docs/testing-overview>`__
+
+Clean Up Tests
+~~~~~~~~~~~~~~
+
+When testing the Realm React Native SDK, you must with 
+:js-sdk:`Realm.close() <react-native-close-a-realm>` after you're done with it to prevent memory leaks. 
+
+You may also want to delete the realm file  with :js-sdk:`Realm.deleteFile() <>`
```
Comment (cbush) [inline source/sdk/react-native/test-and-debug.txt:33]: "may also want to" feels a bit soft. Should we just say "You should"?

File: source/sdk/react-native/test-and-debug.txt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+Realm provides differing levels of support for various React Native debuggers. 
+
+.. seealso::
+
+   `Official React Native Debugging Documentation <https://reactnative.dev/docs/debugging>`__
+
+Safari Developer Tools
+~~~~~~~~~~~~~~~~~~~~~~
+
+If you're a MacOS user, you can use Safari Developer Tools to debug the iOS version
```
Comment (cbush) [inline source/sdk/react-native/test-and-debug.txt:54]:
```suggestion
If you're a macOS user, you can use Safari Developer Tools to debug the iOS version
```

**Review by cbush — APPROVED** [review]

File: source/sdk/react-native/test-and-debug.txt (hunk 3 above)
Comment (mongodben) [reply source/sdk/react-native/test-and-debug.txt:54]: TIL

**Pull request merged**